### PR TITLE
fix: Add a working `cuda` feature

### LINKED ISSUE #259: CUDA acceleration always enabled regardless of `cuda` feature enabled
There have been multiple reports of consistent GPU utilization shown with `nvtop` on Nvidia machines, regardless of if the `cuda`/`opencl` features are enabled. However, there should be no CUDA usage if the feature isn't enabled.

According to @winston-h-zhang this may be due to the following issue in https://github.com/lurk-lab/grumpkin-msm and https://github.com/supranational/pasta-msm:
> build.rs will always enable cuda if it detects a valid nvcc

### PR DESCRIPTION
**Pull request by winston-h-zhang**
Resolves https://github.com/lurk-lab/arecibo/issues/259 downstream. This PR adds an actual `cuda` feature that can be toggled `grumpkin-msm`, and thus all of its dependents. We also add GPU CI checks.

### CODE REVIEW

File: .github/workflows/gpu-ci.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Runs the test suite on a self-hosted GPU machine with CUDA enabled
```
Comment (samuelburnham) [inline .github/workflows/gpu-ci.yml:1]: Nit: we're only building here, not testing. Is the latter intended for later or not a priority?

**Review by samuelburnham — APPROVED** [review]

**Review by huitseeker — CHANGES_REQUESTED** [review]
1. do we really need to push the semantics of the cuda feature inside grumpkin-msm or can we simply not import grumpkin-msm in dependents if that feature is not activated?
2. can we make grumpkin-msm print an **indiscutably spectacular** message if it's loaded and does not manage to resolve to a GPU-using MSM? I really want the fallback to the CPU to be extremely noticeable, 100% of the time, every time.

**Review by huitseeker — CHANGES_REQUESTED** [review]
To reproduce the comment from https://github.com/lurk-lab/arecibo/issues/259, this is going in a good direction given the fresh benchmarks results, but:

we should make grumpkin-msm:
1/ always compile the CPU method, as it does now,
2/ only attempt to compile and run the GPU method when the cuda feature is enabled
3/ emit an indiscutably spectacular message when it's compiled with the cuda feature and does not find the CPU,

**Comment by winston-h-zhang** [comment]
This is the error message:
```
paperspace@psrjp5jvivu1:~/lurk-lab/grumpkin-msm$ cargo build --features cuda
   Compiling sppark v0.1.5
   Compiling grumpkin-msm v0.1.0 (/home/paperspace/lurk-lab/grumpkin-msm)
warning: grumpkin-msm@0.1.0: feature "cuda" was enabled but no valid installation of CUDA was found
warning: grumpkin-msm@0.1.0: the crate's default CPU methods will be compiled instead; NO GPU IMPLEMENTATION WILL BE USED WHEN CALLING THIS CRATE'S METHODS
warning: grumpkin-msm@0.1.0: please recompile without feature "cuda" or provide a valid CUDA installation
    Finished dev [unoptimized + debuginfo] target(s) in 2.48s
```

**Review by huitseeker — APPROVED** [review]
Excellent, thanks a lot!

**Pull request merged**